subprojects/core/src/main/java/org/gradle/api/internal/file/copy/DefaultCopySpec.java

```diff
@@ -523,10 +530,8 @@ public String getFilteringCharset() {
     }
 
     @Override
-    public void setFilteringCharset(@Nullable String charset) {
-        if (charset == null) {
-            throw new InvalidUserDataException("filteringCharset must not be null");
-        }
+    public void setFilteringCharset(String charset) {
+        Preconditions.checkNotNull(charset, "filteringCharset must not be null");
```

Comment:
Now a different exception is thrown (NullPointer vs InvalidUserDataException), so a few unit tests break.